src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -481,6 +481,84 @@ public Consumer<Message<String>> consumeSensitivityAnalysisFailed() {
         };
     }
 
+    @Bean
+    public Consumer<Message<String>> consumeNonEvacuatedEnergyResult() {
+        return message -> {
+            UUID resultUuid = UUID.fromString(message.getHeaders().get(RESULT_UUID, String.class));
+            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
+            if (receiver != null) {
+                NodeReceiver receiverObj;
+                try {
+                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                    LOGGER.info("Sensitivity analysis non evacuated energy result '{}' available for node '{}'", resultUuid, receiverObj.getNodeUuid());
+
+                    // update DB
+                    updateNonEvacuatedEnergyResultUuid(receiverObj.getNodeUuid(), resultUuid);
+
+                    // send notifications
+                    UUID studyUuid = networkModificationTreeService.getStudyUuidForNodeId(receiverObj.getNodeUuid());
+
+                    notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), NotificationService.UPDATE_TYPE_NON_EVACUATED_ENERGY_STATUS);
+                    notificationService.emitStudyChanged(studyUuid, receiverObj.getNodeUuid(), NotificationService.UPDATE_TYPE_NON_EVACUATED_ENERGY_RESULT);
+                } catch (JsonProcessingException e) {
+                    LOGGER.error(e.toString());
+                }
+            }
+        };
+    }
+
+    @Bean
+    public Consumer<Message<String>> consumeNonEvacuatedEnergyStopped() {
+        return message -> {
+            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
+            if (receiver != null) {
+                NodeReceiver receiverObj;
+                try {
+                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                    LOGGER.info("Sensitivity analysis non evacuated energy stopped for node '{}'", receiverObj.getNodeUuid());
+
+                    // delete sensitivity analysis non evacuated energy result in database
+                    updateNonEvacuatedEnergyResultUuid(receiverObj.getNodeUuid(), null);
```

Comment:
Is there already a result if the computation is stopped ?

Reply:
Same thing is done for all other computations ... it is to ensure that there is not result associated to the node